parser.js

```diff
@@ -11,6 +11,16 @@ function makeUrlAbsolute(base, relative) {
   return relative;
 }
 
+function getProvider(url) {
+  return urlparse.parse(url)
+    .hostname
+    .replace(/www[a-zA-Z0-9]*\./, '')
+    .replace('co.', '')
```

Comment:
The special casing of this one particular second-level TLD is mildly concerning.
Not sure how many others (if any), we should bother to handle: https://en.wikipedia.org/wiki/Second-level_domain

But probably can call it "good enough" for now, and address in the future if we notice weird results.

```js
function getProvider(hostname) {
  const out = hostname
    .replace(/www[a-zA-Z0-9]*\./, '')
    .replace('co.', '')
    .split('.')
    .slice(0, -1)
    .join(' ');
  console.log(out);
}

getProvider('yahoo.com'); // "yahoo"
getProvider('www.reddit.com'); // "reddit"
getProvider('sports.bing.com'); // "sports bing"
getProvider('www.bbc.co.uk'); // "bbc"
getProvider('www.cnn.com'); // "cnn"
getProvider('aeon.co'); // "aeon"
getProvider('mobile.nytimes.com'); // "mobile nytimes"
getProvider('www1.foobar.ca'); // "foobar"
getProvider('mobile.www.barfoo.ninja'); // "mobile barfoo" // <== removes "www." from middle of URL.
getProvider('www.school.k12.il'); // "school k12"
```